### PR TITLE
Add useProgress tests

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | We need to test the basic usage of the useProgressRing component
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI should be enough

### CODE REVIEW

**Comment by NeOMakinG** [comment]
@mparvazi for review

I think this component doesn't require more tests, it all about basic usage, the interface isn't that complex

File: src/js/constants/mocks/useProgressRing-data.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  <svg class="progress-ring text-success col-4" width="74" height="74" style="width: 74px; height: 74px;">
+    <circle class="progress-ring__background-circle" stroke-width="4" fill="transparent" r="29" cx="37" cy="37"></circle>
+    <circle class="progress-ring__circle" stroke="currentColor" stroke-width="4" data-percent="75" fill="transparent" r="29" cx="37" cy="37" style="stroke-dasharray: 182.212, 182.212; stroke-dashoffset: 45.5531;"></circle>
+  </svg>
+ `;
+
+export default {
+  Template,
+  TemplateWithoutText,
+};
```
Comment (mparvazi) [inline src/js/constants/mocks/useProgressRing-data.ts:46]:
```suggestion
export const Template = `
  <svg class="progress-ring text-success col-4" width="74" height="74" style="width: 74px; height: 74px;">
    <circle
      class="progress-ring__background-circle" 
      stroke-width="4" 
      fill="transparent"
      r="29"
      cx="37" 
      cy="37"
    ></circle>
    <circle 
      class="progress-ring__circle" 
      stroke="currentColor" 
      stroke-width="4" 
      data-percent="0" 
      fill="transparent" 
      r="29" 
      cx="37" 
      cy="37"
      style="stroke-dasharray: 182.212, 182.212; stroke-dashoffset: 0;"
    ></circle>
    <text x="50%" y="50%" dominant-baseline="middle" text-anchor="middle"></text>
  </svg>
 `;

export const TemplateWithoutText = `
  <svg class="progress-ring text-success col-4" width="74" height="74" style="width: 74px; height: 74px;">
    <circle
      class="progress-ring__background-circle" 
      stroke-width="4" 
      fill="transparent"
      r="29"
      cx="37" 
      cy="37"
    ></circle>
    <circle 
      class="progress-ring__circle" 
      stroke="currentColor" 
      stroke-width="4" 
      data-percent="0" 
      fill="transparent" 
      r="29" 
      cx="37" 
      cy="37"
      style="stroke-dasharray: 182.212, 182.212; stroke-dashoffset: 0;"
    ></circle>
  </svg>
 `;

export const TemplateWithoutCircle = `
 <svg class="progress-ring text-success col-4" width="74" height="74" style="width: 74px; height: 74px;">
   <text x="50%" y="50%" dominant-baseline="middle" text-anchor="middle"></text>
 </svg>
`;

export default {
  Template,
  TemplateWithoutText,
};

```

File: src/js/components/useProgressRing.test.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      if (!setProgress) return;
+
+      setProgress(4);
+
+      const text = document.querySelector(ProgressRingMap.text);
+
+      expect(text?.innerHTML).toBeUndefined();
+    });
+  });
+});
```
Comment (mparvazi) [inline src/js/components/useProgressRing.test.ts:101]:
```suggestion
describe('useProgressRing', () => {
  describe('with valid template', () => {
    beforeAll(() => {
      resetHTMLBodyContent(ProgressRing.Template);
      window.prestashop = {};
      initEmitter();
    });

    it('should display an empty text when setProgress not used', async () => {
      const complete = 4;
      useProgressRing(ProgressRingMap.checkout.element, {steps: complete});
      const text = document.querySelector(ProgressRingMap.text);

      expect(text?.innerHTML).toBe('');
    });

    it('should be on specefic step after setProgress', async () => {
      const complete = 4;
      const current = 3;
      const {setProgress} = useProgressRing(ProgressRingMap.checkout.element, {steps: complete});

      if (setProgress) {
        setProgress(current);
      }

      const text = document.querySelector(ProgressRingMap.text);

      expect(text?.innerHTML).toBe(`${current} / ${complete}`);
    });

    it('should be in percent style when is selected', async () => {
      const complete = 15;
      const current = 7;
      const percentage = Math.floor((current / complete) * 100);
      const {setProgress} = useProgressRing(
        ProgressRingMap.checkout.element,
        {steps: complete, text: ProgressRingText.percent},
      );

      if (setProgress) {
        setProgress(current);
      }

      const text = document.querySelector(ProgressRingMap.text);

      expect(text?.innerHTML).toBe(`${percentage}%`);
    });
  });

  describe('without valid template', () => {
    beforeAll(() => {
      window.prestashop = {};
      initEmitter();
    });

    it('should display a progress ring without any text when using setProgress', async () => {
      resetHTMLBodyContent(ProgressRing.TemplateWithoutText);
      const complete = 4;
      const {setProgress} = useProgressRing(ProgressRingMap.checkout.element, {steps: complete});

      if (setProgress) {
        setProgress(4);
      }

      const text = document.querySelector(ProgressRingMap.text);

      expect(text?.innerHTML).toBeUndefined();
    });

    it('should return undefiend setProgress when the circle is not in the tempalte', async () => {
      resetHTMLBodyContent(ProgressRing.TemplateWithoutCircle);
      const complete = 4;
      const {setProgress} = useProgressRing(ProgressRingMap.checkout.element, {steps: complete});

      expect(setProgress).toBeUndefined();
    });
  });

  describe('without template', () => {
    it('should return an error message', async () => {
      resetHTMLBodyContent('');
      const complete = 4;
      const {error} = useProgressRing(ProgressRingMap.checkout.element, {steps: complete});

      expect(error).toBeDefined();
    });
  });
});

```

**Review by mparvazi — COMMENTED** [review]
`src/js/components/useProgressRing.ts`

Screenshot from 2022-05-04 06-47-37

File: src/js/components/useProgressRing.test.ts (hunk 2 above)
Comment (mparvazi) [reply src/js/components/useProgressRing.test.ts:101]: 100% coverage.

Comment (mparvazi) [reply src/js/components/useProgressRing.test.ts:101]:
```
if (!setProgress) return;
setProgress(86);
```
must be:

```
if (setProgress) {
   setProgress(current);
}
```

We should not break test block.

**Comment by NeOMakinG** [comment]
@mparvazi I reverted the `Math.floor` as it's not needed, it's just used to display the filled ring

**Comment by NeOMakinG** [comment]
@mparvazi if you feel like it's mergeable, provide a stamp on this PR, I'll merge it right after

**Comment by mparvazi** [comment]
I've tested, it's ready to merge.

**Comment by NeOMakinG** [comment]
Thanks @mparvazi

**Pull request merged**